server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -577,22 +577,21 @@ public void test_prune_outputs_on_group_hash_aggregate() {
             UNION
             SELECT a::int ai, avg(x), i::long FROM t1 WHERE a='3' GROUP BY 1,3
             """);
-        assertThat(plan).isEqualTo(
-            """
-            GroupHashAggregate[ai, "avg(x)", "cast(i AS bigint)"]
-              └ Union[ai, "avg(x)", "cast(i AS bigint)"]
-                ├ GroupHashAggregate[ai, "avg(x)", "cast(i AS bigint)"]
-                │  └ Union[ai, "avg(x)", "cast(i AS bigint)"]
-                │    ├ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]
-                │    │  └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]
-                │    │    └ Collect[doc.t1 | [x, cast(a AS integer) AS ai, cast(i AS bigint)] | (a = '1')]
-                │    └ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]
-                │      └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]
-                │        └ Collect[doc.t1 | [x, cast(a AS integer) AS ai, cast(i AS bigint)] | (a = '2')]
-                └ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]
-                  └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]
-                    └ Collect[doc.t1 | [x, cast(a AS integer) AS ai, cast(i AS bigint)] | (a = '3')]
-            """);
+        assertThat(plan).hasOperators(
+            "GroupHashAggregate[ai, \"avg(x)\", \"cast(i AS bigint)\"]",
+            "  └ Union[ai, \"avg(x)\", \"cast(i AS bigint)\"]",
+            "    ├ GroupHashAggregate[ai, \"avg(x)\", \"cast(i AS bigint)\"]",
+            "    │  └ Union[ai, \"avg(x)\", \"cast(i AS bigint)\"]",
+            "    │    ├ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]",
+            "    │    │  └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]",
+            "    │    │    └ Collect[doc.t1 | [cast(a AS integer) AS ai, cast(i AS bigint), x] | (a = '1')]",
+            "    │    └ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]",
+            "    │      └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]",
+            "    │        └ Collect[doc.t1 | [cast(a AS integer) AS ai, cast(i AS bigint), x] | (a = '2')]",
+            "    └ Eval[cast(a AS integer) AS ai, avg(x), cast(i AS bigint)]",
+            "      └ GroupHashAggregate[cast(a AS integer) AS ai, cast(i AS bigint) | avg(x)]",
+            "        └ Collect[doc.t1 | [cast(a AS integer) AS ai, cast(i AS bigint), x] | (a = '3')]"
```

Comment:
The order of the collect operators is now based on the order of the outputs of the top final select from the whole query.

Reply:
Not sure whether it could be a problem in a mixed cluster: for example merging results from a node without the fix and node with the fix. 

But AFAIK we never account for that when we do some fixes, otherwise even pruning optimizations would not be possible without minVerson check. Or maybe I'm missing something and such query will fail with some meaningful error.

Reply:
Logical planning does not happen distributed on each node, only the execution plan will be distributed.

Reply:
Thanks for the clarification!